ui/app/hooks/useTokenDisplayValue.js

```diff
@@ -0,0 +1,38 @@
+import { getTokenValue, calcTokenAmount } from '../helpers/utils/token-util'
+import { getTokenData } from '../helpers/utils/transactions.util'
+import { useMemo } from 'react'
+
+/**
+ * Defines the shape for the Token input parameter for useTokenDisplayValue
+ * @typedef {Object} Token
+ * @property {string} symbol   - The string to use as a suffix for the token (eg. DAI)
+ * @property {decimals} number - The number of decimals to show when displaying this type of token
+ */
+
+/**
+ * useTokenDisplayValue
+ * Given the data string from txParams and a token object with symbol and decimals, return
+ * a displayValue that represents a string representing that token amount as a string. Also
+ * return a tokenData object for downstream usage and the suffix for the token to use as props
+ * for other hooks and/or components
+ * @param {string} transactionData
+ * @param {Token} token
+ * @return {[string, Object]} - a tuple with displayValue as first item and tokenData as second
+ */
+export function useTokenDisplayValue (transactionData, token) {
+  if (!transactionData || !token) {
+    return null
+  }
+  const tokenData = useMemo(() => getTokenData(transactionData), [transactionData])
+  if (!tokenData?.params?.length) {
+    return null
+  }
+  const { decimals } = token
+
+  const displayValue = useMemo(() => {
+    const tokenValue = getTokenValue(tokenData.params)
+    return calcTokenAmount(tokenValue, decimals).toString()
+  }, [tokenData, decimals])
+
+  return [displayValue, tokenData]
```

Comment:
Is the second entry, `tokenData`, intended for some future use? It doesn't appear to be used at the moment.

Reply:
my goal was to provide a way for consumers to act on token data as well without needing to also decode it. I suppose useTokenData could totally be a separate hook, though multiple consumers decoding the same data string might be bad? Maybe its a quick operation and thinking of the performance impact of this:

```js
const tokenData = useTokenData(data)
const tokenDisplayValue = useTokenDisplayValue(data, token) // <- also calls useTokenData
```

is just premature optimization. alternatively, we can require useTokenDisplayValue to be passed an already decoded data string. 

Thoughts @Gudahtt ?

Reply:
Returning the token data seems sensible enough, sure. Just that nobody was using it yet, and even in the PR based upon this it isn't used for anything.

Reply:
Removed the tokenData return value, will add it back in the other PR or think of a better way of consuming this.